pkg/monitoring/prometheusagent/service.go

```diff
@@ -189,14 +189,13 @@ func (pas PrometheusAgentService) deleteConfigMap(ctx context.Context, cluster *
 	}
 
 	// Delete the finalizer
-	desired := current.DeepCopy()
```

Comment:
How does removing the `DeepCopy` fixes the finalizer removal /

Reply:
Deepcopy is useless if we are updating instead of patching

Reply:
In both case `DeepCopy` should not be needed, since the object is local to the function.

But how does moving from patch to update fixes the issue then ?

Reply:
I was advised by upstream to inverse the order of the delete and the patch, that's worth trying

Reply:
Testing here https://github.com/giantswarm/observability-operator/pull/32